miden-lib/asm/note_scripts/basic.masm

```diff
@@ -0,0 +1,75 @@
+    
+    assert_eq                                           # [] if account_id == target_account_id, fails if not
+
+    push.0                                              # [0, ...] memory pointer to store assets
+    exec.note::get_assets                               # [num_of_assets, 0, ...] 
+
+    dup neq.0                                           # [1 || 0, num_of_assets, 0, ...], sanity check that there is at least one asset
+    while.true                                          
+        sub.1 padw swap.5 dup swap.6                    # [num_of_assets - (i), 0, 0, 0, 0, num_of_assets - (i), ...], i ~ round in the loop
+        mem_loadw                                       # [ASSET, num_of_assets - (i), ...], i ~ round in the loop
+        call.wallet::receive_asset dropw                # [num_of_assets - (i), ...], i ~ round in the loop
+        dup neq.0                                       # [1 || 0, num_of_assets - (i), ...], if end of memory reached, break
+    end
+
+    drop drop
+end
+
+#! Pay to ID reclaimable - ensures that note can only be consumed by an account with the provided ID
+#! or after provided block number only by the sender account
+#! 
+#! Inputs: # [reclaim_block_height, target_account_id, sender_account_id, ...]
+#! Outputs: []
+#!
+#! - ASSET is the asset to be received, can be fungible or non-fungible
+#!
+#! FAILS if: 
+#! - Account ID of executing account is not equal to specified Account ID
+#! - The same non-fungible asset already exists in the account.
+#! - Adding a fungible asset would result in amount overflow, i.e.,
+#!   the total amount would be greater than 2^63.
+export.p2idr                                            # [note_inputs = target_account_id, ...]                                            
+    exec.tx::get_block_number                           # [current_block_height, reclaim_block_height, target_account_id, sender_account_id, ...]    
+    lte                                                 # [1 || 0, target_account_id, sender_account_id, ...], 1 if current_block_height >= reclaim_block_height, 0 otherwise
+
+    if.true                                             # current_block_height >= reclaim_block_height, [target_account_id, sender_account_id, ...]
+        drop                                            # [sender_account_id, ...]
+        exec.account::get_id                            # [account_id, sender_account_id, ...]
+        assert_eq                                       # [] if account_id == sender_account_id, fails if not                                                   
+    else                                                # current_block_height < reclaim_block_height, [target_account_id, sender_account_id, ...]
+        exec.account::get_id                            # [account_id, target_account_id, sender_account_id, ...]
+        assert_eq                                       # [] if account_id == target_account_id, fails if not
+    end
```

Comment:
I think this logic is slightly different from what I had in mind. The original idea was:

- Before `reclaim_block_height` only `target_account` can consume then note.
- After `reclaim_block_height` either `target_account` or `sender_account` would be able to consume it.

I believe the difference right now is that after `reclaim_block_height` only the sender can consume it.

Reply:
Jup, that is correct. The question is, do we want to allow the `target_account` to be able to consume the note at the same time as the `sender_account`? 

I can change it accordingly, but there are pros and cons, right? It might end up in a race condition.